tests/testsuite/tree.rs

```diff
@@ -1085,6 +1085,57 @@ fn duplicates_with_target() {
     p.cargo("tree -d --target=all").with_stdout("").run();
 }
 
+#[cargo_test]
+fn duplicates_with_proc_macro() {
+    Package::new("cat", "1.0.0").publish();
+    Package::new("cat", "2.0.0").publish();
+    Package::new("dep", "1.0.0")
+        .proc_macro(true)
+        .dep("cat", "1.0")
+        .publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "0.1.0"
+
+            [dependencies]
+            dep = "1.0"
+            cat = "2.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("tree")
+        .with_stdout(
+            "\
+foo v0.1.0 ([..]/foo)
+├── cat v2.0.0
+└── dep v1.0.0 (proc-macro)
+    └── cat v1.0.0
+",
+        )
+        .run();
+
+    p.cargo("tree -d")
```

Comment:
nit: can you prefer long flags in the tests, rather than short flags?  For example, I don't run `cargo tree` often enough and had to go and consult the help to verify what `-d` is.

(yes, existing tests do it but we should probably clean that up)